### PR TITLE
[WIP] add LearningTransport

### PR DESCRIPTION
**Pull request by SimonCropp**
WIP until 6.3 is released

### CODE REVIEW

File: nservicebus/transports/index.md [hunk 1]
```diff
@@ -15,6 +15,11 @@ NServiceBus is built on top of existing queuing technologies. In NServiceBus the
 NServiceBus transports can be divided into several categories.
 
 
+### Learning Transport
```
Comment (mikeminutillo) [inline nservicebus/transports/index.md]: I would move this to the bottom.

File: nservicebus/transports/index.md [hunk 2]
```diff
@@ -15,6 +15,11 @@ NServiceBus is built on top of existing queuing technologies. In NServiceBus the
...
+
+The [Learning Transport](/nservicebus/learning-transport/) is for education purposes only and does not fit into the below categories.
```
Comment (mikeminutillo) [inline nservicebus/transports/index.md]: Isn't technically a Unicast, Brokered transport?

File: nservicebus/learning-transport/learning-transport-warning.include.md [hunk 3]
```diff
@@ -0,0 +1 @@
+DANGER: The [Learning Transport](/nservicebus/learning-transport/) is not for production use. It is designed for short term learning and experimentation purposes. It should also not be used as a longer-term development transport, i.e. the same transport used in production should be used in development and debug scenarios. Do not use this transport to perform any kind of performance analysis.
```
Comment (mikeminutillo) [inline nservicebus/learning-transport/learning-transport-warning.include.md]: How long is longer term? Does the statement

> the same transport used in production should be used in development and debug scenarios.

cover it? Should we say something like

> Select a production transport before developing features.

File: nservicebus/learning-transport/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+---
+title: Learning Transport
+component: LearningTransport
+reviewed: 2017-05-01
+---
+
+The Learning Transport stores all message actions in the local file system relative to the current solution directory.
```
Comment (mikeminutillo) [inline nservicebus/learning-transport/index.md]: > The Learning Transport simulates queuing infrastructure by storing all message actions in the local file system.

I think opening with "simulates" or something similar helps to hammer home the non-production nature of it.

File: nservicebus/learning-transport/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Currently [ServiceControl](/servicecontrol/) (and hence [ServicePulse](/servicepulse/) and [ServiceInsight](https://docs.particular.net/serviceinsight/)) are not supported.
+
+
+## Usage
+
+snippet: LearningTransport
+
+
+## File System Structure
```
Comment (mikeminutillo) [inline nservicebus/learning-transport/index.md]: Out of interest, can I lock these files if I open them in a text editor while the system is running? What happens?

File: samples/learning-transport/sample.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+
+include: learning-transport-warning
+
+This sample show how to use the Learning Transport and how the messages are persisted to the file system.
+
+
+## Sample Structure
+
+The sample has two endpoint, `Endpoint1` and `Endpoint2`. Both endpoitns are configured to use the Learning Transport:
```
Comment (mikeminutillo) [inline samples/learning-transport/sample.md]: endpoints

There are two cases that are wrong.

File: nservicebus/learning-transport/index.md (hunk 5 above)
Comment (SimonCropp) [inline nservicebus/learning-transport/index.md]: yes u would get errors.

**Pull request merged**